components/org.wso2.config.diff.creater/src/main/java/org/wso2/config/diff/creater/OutputGenerator.java

```diff
@@ -0,0 +1,170 @@
+    private File outputCSV = new File(MigrationConstants.OUTPUT_FOLDER.concat(MigrationConstants.FILE_SEPARATOR)
+            .concat(MigrationConstants.OUTPUT_CATALOG_CSV));
+    private File outputKeyValues = new File(MigrationConstants.OUTPUT_FOLDER.concat(MigrationConstants.FILE_SEPARATOR)
+            .concat(MigrationConstants.KEY_VALUE_CSV));
+    private File outputDirectory = new File(MigrationConstants.OUTPUT_FOLDER);
+    private File unTemplatedFileDirectory = new File(MigrationConstants.OUTPUT_FOLDER.concat(MigrationConstants
+            .FILE_SEPARATOR).concat(MigrationConstants.UN_TEMPLATE_FILE_FOLDER));
+
+    /**
+     * Create all the output csv files and directory.
+     *
+     * @throws ConfigMigrateException ConfigMigrateException.
+     */
+    public void createOutputFiles() throws ConfigMigrateException {
+
+        try {
+            createDirectory(outputDirectory);
+            createDirectory(unTemplatedFileDirectory);
+
+            createFile(outputCSV);
+            updateFile(outputCSV, MigrationConstants.CATALOG);
+
+            createFile(outputKeyValues);
+            updateFile(outputKeyValues, MigrationConstants.KEY_VALUES);
+        } catch (IOException e) {
+            throw new ConfigMigrateException("Error occurred when creating output files.");
+        }
+    }
+
+
+
+    /**
+     * Create a directory to put un-template customized config files in the migrated setup.
+     *
+     * @param outputDirectory create directory.
+     * @throws IOException exception when director is not created.
+     */
+    private void createDirectory(File outputDirectory) throws IOException {
+
+        if (outputDirectory.exists()) {
+            FileUtils.deleteDirectory(outputDirectory);
+        }
+        if (!outputDirectory.mkdir()) {
+            log.error("The output directory, " + outputDirectory.getName() + " is not created.");
+            throw new IOException("The output directory, " + outputDirectory.getName() + " is not created.");
+        }
+        setOutputDirectory(outputDirectory);
+    }
+
+    /**
+     * Create new file after removing the existing one.
+     * @param file file
+     * @throws IOException Exception is thrown if file is not deleted and created.
+     */
+    private void createFile(File file) throws IOException {
+
+        if (file.exists()) {
+            if (!file.delete()) {
+                log.error("The output file, " + file.getName() + " is not deleted.");
+                throw new IOException("The output file, " + file.getName() + " is not deleted.");
+            }
+        }
+        if (!file.createNewFile()) {
+            log.error("The output file, " + file.getName() + " is not created.");
+            throw new IOException("The output file, " + file.getName() + " is not created.");
+        }
+    }
+
+    /**
+     * Update the created file using the title row of the output file.
+     * @param file filename.
```

Comment:
Would be better to run through the code style again as there are slight style changes needed.